fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalIcebergScanOperator.java

```diff
@@ -65,9 +65,11 @@ public void setScanOperatorPredicates(ScanOperatorPredicates predicates) {
 
     @Override
     public boolean isEmptyOutputRows() {
-        return !table.isUnPartitioned() &&
-                !(((IcebergTable) table).hasPartitionTransformedEvolution()) &&
-                predicates.getSelectedPartitionIds().isEmpty();
+        IcebergTable icebergTable = (IcebergTable) table;
+        return !icebergTable.isUnPartitioned() &&
+               !icebergTable.hasNullPartitionField() &&
```

Comment:
Have you made any partition evolution? When the partition value is null, it's usually accompanied by partition evolution

Reply:
To be honest, It's not a normal iceberg table. This is a very complex and long-standing, and very important table, probably partition evolution.

The query results of presto and spark sql are correct, and sr returns empty for any query.

Reply:
wha't the partition transform type in the corresponding spec id of iceberg native table?

Reply:
Seems like no evolution.

- partition-spec:
```
{
    "name" : "batch_date",
    "transform" : "identity",
    "source-id" : 58,
    "field-id" : 1000
  }, {
    "name" : "user_id_bucket_500",
    "transform" : "bucket[500]",
    "source-id" : 4,
    "field-id" : 1001
  }, {
    "name" : "batch_hour_500",
    "transform" : "identity",
    "source-id" : 59,
    "field-id" : 1002
  }, {
    "name" : "batch_minute_500",
    "transform" : "identity",
    "source-id" : 60,
    "field-id" : 1003
  }
default-spec-id: 0
```
- manifest file:
  ```
"data_file": {
    "file_path": xxx.parquet",        
"file_format": "PARQUET",
    "partition": {
      "batch_date": {
        "string": "20240106"
      },
      "user_id_bucket_500": {
        "int": 319
      },
      "batch_hour_500": null,
      "batch_minute_500": null
    },
```

Reply:
so, maybe you use api to sink iceberg data. otherwise, the partition transformed type will be VoidTransform.